app/views/decidim/proposals/proposals/show.html.erb

```diff
@@ -28,6 +28,24 @@
       <% if component_settings.geocoding_enabled? %>
         <%= render partial: "decidim/shared/static_map", locals: { icon_name: "proposals", geolocalizable: @proposal } %>
       <% end %>
+      <% if @proposal.equity_composite_index_percentile != nil %>
+        <figure>
+          <figcaption class="text-uppercase">
+            <strong>
+              <%= t("proposal_title", scope: "decidim.proposals.equity") %>
+            </strong>
+          </figcaption>
+          <ul>
+            <li>
+              <% equity_quintile = (@proposal.equity_composite_index_percentile.round(2) * 5).floor() + 1 %>
```

Comment:
I think having this computation in the template is pretty gross, but 1) I couldn't figure out which file is responsible for populating the data in this template? and 2) if that file is in another module, we'd have to duplicate it, which is probably more gross.

Reply:
Yeah. It's feels pretty low-level.

This project uses presenters, so I think the normal place to put this sort of calculation is in the proposal presenter, but yeah, you'd have to duplicate it. Once you did that you could do this:

```<%= present(@proposal).equity_quintile %>```

Another option is to create a classic rails view helper: app/helpers/decidim/proposals/proposal_helper.rb:

```
module Decidim
  module Proposals
    module ProposalHelper
      def calculate_quintile(percentile)
        (percentile.round(2) * 5).floor() + 1
      end
    end
  end
end
```

then in the view `<% equity_quintile = calculate_quintile(@proposal.equity_composite_index_percentile) %>`

Or just leave it. It's up to you.

Reply:
I'll follow up on this in a future PR. Good suggestion!